packages/sdk/src/fee/FeeEstimator.ts

```diff
@@ -0,0 +1,142 @@
+import { Psbt } from "bitcoinjs-lib";
+import { Buffer } from "buffer";
+
+import type { Network } from "../config/types";
+import { MAXIMUM_FEE } from "../constants";
+import { OrditSDKError } from "../errors";
+import { getNetwork, getScriptType } from "../utils";
+import { getBaseSizeByType, TRANSACTION_HEADER_SIZE } from "./helper";
+import type { FeeEstimatorOptions } from "./types";
+
+class FeeEstimator {
+  protected fee = 0;
+
+  protected feeRate: number;
+
+  protected network: Network;
+
+  protected psbt: Psbt;
+
+  protected witness?: Buffer[] = [];
+
+  protected virtualSize = 0;
+
+  protected weight = 0;
+
+  constructor({ feeRate, network, psbt, witness }: FeeEstimatorOptions) {
+    // feeRate can be 0 because a seller does not pay for network fees
+    if (feeRate < 0 || !Number.isSafeInteger(feeRate)) {
+      throw new OrditSDKError("Invalid feeRate");
+    }
+
+    this.feeRate = feeRate;
+    this.network = network;
+    this.witness = witness || [];
+    this.psbt = psbt || new Psbt({ network: getNetwork(this.network) });
+  }
+
+  get data() {
+    return {
+      fee: this.fee,
+      virtualSize: this.virtualSize,
+      weight: this.weight,
+    };
+  }
+
+  calculateNetworkFee(): number {
+    this.fee = this.calculateVirtualSize() * this.feeRate;
+    // Prevents catastrophic calculations from happening
+    if (this.fee > MAXIMUM_FEE) {
+      throw new OrditSDKError("Error while calculating fees");
+    }
+
+    return this.fee;
+  }
```

Comment:
i think SDK should not dictate max network fee? 🤔 we might need a better safe guard than this.

Reply:
Let's resolve thjis on another PR, I think this PR getting too long to merge now - #48

Reply:
cc: @Nanosync